src/parser/jobs/ast/modules/ArcanaTracking/ArcanaSuggestions.tsx

```diff
@@ -56,172 +48,132 @@ interface SleeveIcon {
-		this.addEventHook('cast', {by: 'player'}, this.onCast)
+		const divinationFilter = filter<Event>().type('statusApply').status(this.data.statuses.DIVINATION.id)
+		//this.addEventHook(divinationFilter.target(this.parser.actor.id), this.tryOpenWindow)
+		this.addEventHook(divinationFilter.source(this.parser.actor.id), this.countDivinationBuffs)
+		this.addEventHook(
+			filter<Event>()
+				.type('statusRemove')
+				.target(this.parser.actor.id)
+				.status(this.data.statuses.DIVINATION.id),
+			this.tryCloseWindow,
+		)
 
 		this.addEventHook('complete', this._onComplete)
 	}
 
 	//functions for divination buff counting
-	private countDIVBuffs(event: NormalisedApplyBuffEvent) {
+	private countDivinationBuffs(event: Events['statusApply']) {
 		// Get this from tryOpenWindow. If a window wasn't open, we'll open one.
-		const lastWindow: DIVWindow | undefined = this.tryOpenWindow(event)
+		// If it was already open (because another Astrologian went first), we'll keep using it
+		this.tryOpenWindow(event)
 
 		// Find out how many players we hit with the buff.
-		if (lastWindow) {
-			lastWindow.playersBuffed += event.confirmedEvents.filter(hit => this.parser.fightFriendlies.findIndex(f => f.id === hit.targetID) >= 0).length
+		if (this.currentWindow != null && !this.currentWindow.playersBuffed.includes(event.target) && this.actors.get(event.target).playerControlled) {
+			this.currentWindow.playersBuffed.push(event.target)
 		}
 	}
 
-	private tryOpenWindow(event: NormalisedApplyBuffEvent): DIVWindow | undefined {
-		const lastWindow: DIVWindow | undefined = _.last(this.history)
-
-		if (lastWindow && !lastWindow.end) {
-			return lastWindow
+	private tryOpenWindow(event: Events['statusApply']): DivinationWindow {
+		if (this.currentWindow === undefined) {
+			this.currentWindow = {
+				start: event.timestamp,
+				buffsRemoved: [],
+				playersBuffed: [],
+			}
 		}
 
-		if (event.sourceID && event.sourceID === this.parser.player.id) {
-			const newWindow = new DIVWindow(event.timestamp)
-
-			// Handle multiple AST's buffs overwriting each other, we'll have a remove then an apply with the same timestamp
-			// If that happens, mark the window and return
-			newWindow.containsOtherAST = this.lastDIVFalloffTime === event.timestamp
-			//TODO use this to track if overwritten. Consider moving this tip to suggestions
-
-			this.history.push(newWindow)
-			return newWindow
+		// Handle multiple Astrologian's buffs overwriting each other, we'll have a remove then an apply with the same timestamp
+		// If that happens, re-open the last window and keep tracking
+		if (this.currentWindow != null) {
+			if (this.currentWindow.end == null) {
+				return this.currentWindow
+			}
+			if (this.currentWindow.end === event.timestamp) {
+				this.currentWindow.end = undefined
+				return this.currentWindow
+			}
 		}
 
-		return undefined
+		return this.currentWindow
```

Comment:
You don't need to return anything here at all.

Reply:
removed return references